src/main/java/ru/ayaz/ChatServer.java

```diff
@@ -0,0 +1,167 @@
+package ru.ayaz;
+
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class ChatServer {
+
+    private ServerSocket serverSocket;
+    private Map<String, User> usersMap;
+    private Map<String, PrintWriter> outputWriterMap;
+
+
+    public ChatServer() {
+        try {
+            this.serverSocket = new ServerSocket(4400);
+            this.usersMap = new TreeMap<String, User>();
+            this.outputWriterMap = new TreeMap<String, PrintWriter>();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    private void sendMessageToEveryone(String message, String sender) {
+
+        System.out.println(message);
+
+        for (Map.Entry<String, User> entry : usersMap.entrySet()) {
+            String key = entry.getKey();
+            User value = entry.getValue();
+            if(!key.equals(sender)) {
+                if (!value.isInIgnoredSet(sender)) {
+                    PrintWriter pw = outputWriterMap.get(key);
+                    pw.println(message);
+                    pw.flush();
+                }
+            }
+        }
+
+    }
+
+
+    public void startChat() {
+
+        try {
+            while (true) {
+                Socket socket = serverSocket.accept();
+                Thread t = new Thread(new UserHandler(socket));
+                t.start();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    public static void main(String[] args) {
+        ChatServer chat = new ChatServer();
+        chat.startChat();
+    }
+
+
+    public class UserHandler implements Runnable {
```

Comment:
I don't like the concept of inner classes very much. E.g. if you want to test it, you have to create the outer class - ChatServer in your case - which is very cumbersome and unnecessary.
Let's try to redesign this thing (ChatServer and UserHandler) using TDD - may be not right now, but in the next iteration (after you fix all other issues and I merge this pull request).